STYLE_GUIDE.md

```diff
@@ -0,0 +1,196 @@
+# Style Guide for Pull Requests
+
+To ensure consistency throughout the codebase and to aid any of you that are
+interested in contributing, please follow these instructions when structuring
+your code:
+
+# Indentation 
+
+The most general rule is to try and keep everything aligned to a multiple of two
+spaces, so we always wrap after `=` and `in`, like so:
+
+```haskell
+let
+  foo =
+    if quux xs0 then
+      unlines xs0
+    else
+      unlines xs1
+in
+  length foo
+```
+
+Note the placement of the `then` immediately after the `if condition`, this
+ensures the two space indentation is maintained because the positive case is
+then indented on the following line.
+
+This applies to data type definitions as well:
+
+```haskell
+data Foo =
+  Foo {
+      fooFieldA :: Type 
+    , fooFieldB :: Type
+    } deriving Show
```

Comment:
Might recommend making the records odd length to show that the types are not aligned in any way

Reply:
good point

Reply:
was about to say this, good point indeed :sun_with_face: :+1:

Reply:
Do you mean the record *field names* an odd length? 

```haskell
data Foo =
  Foo {
      fooFieldA :: Type 
    , fooB :: Type
    } deriving (Show)
```

Reply:
No, I think @nhibberd means just

```haskell
data Foo =
  Foo {
      fooFieldA :: Type 
    , fooFieldB :: Type
    } deriving (Show)
```

so that the comma `,` emphasizes the reason `fooFieldA` is indented this way.

Reply:
Both, the record field names as odd length shows:

```haskell
data Foo =
  Foo {
      fooFieldA :: Type 
    , fooB :: Type
    } deriving (Show)
```

As opposed to some haskell styles where it might be:

```haskell
data Foo =
  Foo {
      fooFieldA :: Type 
    , fooB      :: Type
    } deriving (Show)
```

The former is the preferred style here if I am not mistaken :)

Reply:
Yeah, @nhibberd is not mistaken :)

Reply:
Awesome!